enhancements/cluster-logging/hypershift-audit-logs.md

```diff
@@ -21,174 +21,124 @@ see-also:
 - Multiple audit policies can be defined in a hosted cluster, independent of the management cluster policies.
   See [api-audit-log-policy](/api-audit-log-policy.md) for how the CLO will support this.
-- Allow some forwarding in the _management_ cluster to be configured from the _hosted_ cluster.
-- Keep direct control of management cluster forwarding on the management cluster.
-  - hosted cluster has no direct access to control plane configuration.
-  - data plane resources can request forwarding by the management cluster.
-  - management cluster HLO automatically validates data-plane requests, and enables them if they are safe.
+- Data plane (customer) requests audit log forwarding by modifying data-plane CLF configuration, forwarding from control plane starts automatically.
 
 ### Non-Goals
 
 - This proposal only covers API audit logs. No change to application, infrastructure or node audit logs.
-- Management cluster is not _required_ to honour all possible forwarding configurations allowed by CLF. \
 
 ## Proposal
 
 ### Workflow Description
 
 #### Hosted Cluster, data plane
 
-1. Create a `Policy.audit.k8s.io` resource with the desired policy
-   (see [api-audit-log-policy](/api-audit-log-policy.md))
-1. Create a `HypershiftLogForwarder` (HLF) resource with an `audit` input linked to the policy.
-1. Create HLF `outputs` and `pipelines` to forward the resulting logs.
-1. Optionally create a `ClusterLogForwarder` to forward other logs (not API-audit) as usual.
+1. Install the Cluster Logging Operator (CLO).
+1. Create a `ClusterLogForwarder` with a HTTP intupt.
+ - Receiver should be identified with a special input name e.g. `hypershiftAPIAudit`
+ - Pipelines from the receiver may include an [api-audit-log-policy](/api-audit-log-policy.md) filter.
+1. Forward logs to any outputs as desired.
 
 #### Management Cluster, control plane
 
-On the _management_ cluster
-1. Install the Cluster Logging Operator (CLO), watching all name-spaces.
-1. Install the new Hypershift Logging Operator (HLO), watching all name-spaces.
-1. In each new HCP name-space, create a `ClusterLogForwarderTemplate` resource. \
-   This resource may be identical for all HCPs or customized per HCP.
+1. Install the Cluster Logging Operator (CLO).
+1. Create a `ClusterLogForwarder` for each HCP to forward to Splunk and management-side CloudWatch.
+1. Install the new Hypershift Logging Operator (HLO) watching each HCP namespace.
+   - The HLO watches CLF configurations _in the data plane_, not the control plane.
+   - When the HLO detects a _control plane_ CLF requesting audit log forwarding, it updates
+     the _data plane_ CLF for that HCP to forward audit logs via a HTTP output to the _control plane_ CLF.
+   - The HLO should copy audit log policies from  _control plane_ CLF pipelines to the _data plane_,
+     to forward only those logs that will also be forwarded by the data plane CLF.
```

Comment:
```suggestion
   - The HLO should copy audit log policies from  _data plane_ CLF pipelines to the _control plane_,
     to forward only those logs that will also be forwarded by the data plane CLF.
```

Customer can set policy in the data plane CLF, it needs copied to the control plane to be applied.

Reply:
Oops again. Note that the policy will be applied on the data plane, it is an optimization (but a very important one) to apply it on the control plane first to reduce the traffic volume. It is also possible for the control plane to apply a default policy of its own to put an upper bound on what the data plane can request. E.g. the data plan could filter out silly stuff like "get -A pods with full response body" so that the data plane will never get that data even if their filter allows it. Need SRE experties to say what the right defaults are, but the audit policy code has defaults inherited from the old splunk forwarder that might be reasonable to impose.

Reply:
@alanconway where are control plane defaults documented?  How are merges between control plane and data plane policy managed?  What messaging is given to customer if their policy is asking for more than what the control plane will allow?  The UX here is important to understand.  And before building this I'd want to identify the use case.  If customers want all audit events, what is the implications to RH if we allow it in the managed context?  Not sure we even have egress costs, but that may be it.  Load on the MC?